ext/mruby_engine/dlmalloc.c

```diff
@@ -4523,6 +4527,40 @@ struct mallinfo mspace_mallinfo(mspace msp) {
   }
   return internal_mallinfo(ms);
 }
+
+void log_stuff(mspace msp) {
+  mstate m = (mstate)msp;
+  ensure_initialization();
+  if (!PREACTION(m)) {
+    check_malloc_state(m);
+    if (is_initialized(m)) {
+      size_t nfree = SIZE_T_ONE; /* top always free */
+      size_t mfree = m->topsize + TOP_FOOT_SIZE;
+      size_t sum = mfree;
+      msegmentptr s = &m->seg;
+      while (s != 0) {
+        fprintf(stderr, "S @ %p\n", (void *)s);
+        mchunkptr q = align_as_chunk(s->base);
+        while (segment_holds(s, q) &&
+               q != m->top && q->head != FENCEPOST_HEAD) {
+          size_t sz = chunksize(q);
+          sum += sz;
+          if (!is_inuse(q)) {
+            mfree += sz;
+            ++nfree;
+          }
+          q = next_chunk(q);
+        }
+        s = s->next;
+      }
+
+      fprintf(stderr, "%zu, %zu, %zu, %zu, %zu, %zu, %zu\n",
+          sum, nfree, m->footprint - sum, m->max_footprint, m->footprint - mfree, mfree, m->topsize);
```

Comment:
You're really against any kind of maths, aren't you? ;)
More seriously though, wrt prefix, I _thought_ I understood this would be part of the "log entry" we currently find by searching for `[SCRIPT]` in splunk, is that right?
I certainly can add the same prefix there, but just wanted to make sure I understand things properly here.